### ISSUE TITLE
read output 

### ISSUE THREAD

**Issue by xuezhicang** [KCallaghan/FlowFill#2]
Hi, I have question about the output results. I know the data type is 32-bit float. After I read the results file, there are 242004 grids in my outputs, but the input only has 245*131 values. How do I figure out the filled DEM.

Could you help me take a look? The files are in my web drive https://drive.google.com/drive/folders/1zTaihXzNT8ox6-_u-AXsZ7hF9R3WsWu0?usp=sharing

@KCallaghan Thank you for your time in advance.

**Comment by KCallaghan** [comment]
Hi, I'm taking a look at your output files now and something doesn't seem right; are you sure that the starting size was 245*131? Could you provide the line or file you used to run this code? (if you used my runfile and user data file, could I see how you filled out the user data file?)

There is some post-processing that you need to do on the output files: FlowFill will add a border one cell wide, but that doesn't seem to be the case here since the size is still off. My preferred way to handle this is to load the data into python, reshape it using the output file dimensions, and shave off the border, i.e.: 
test = np.fromfile('my_outfile.dat',dtype = np.float32)
test = test.reshape(247,-1)  (if the starting size of the file was 245)
test = test[:-2,1:-1]

You can then save as a .dat or .nc, etc, or just use python inside GRASS to do this if you wanted to use GRASS for further processing.

**Comment by xuezhicang** [comment]
Thank you for your reply. 

I have uploaded my input data and my results to google drive. The link: https://drive.google.com/drive/folders/1H0hzochY0xo37FSlkaCqusMa0sA_NPR8?usp=sharing

The "test_netcdf_2.nc" is my input data.
The "test_code" is the compiled file.

When I run this depression filling program, firstly I modify the "user_inputs" file (change the name of input data and the numbers of row and column), then I run "bash run_me.sh". Is it the correct way?

Thank you for your time in advance. @KCallaghan

**Comment by KCallaghan** [comment]
Thanks, that helps. There are a couple of problems in the user_inputs file. Columns and rows should reflect the actual dimensions of your data, so rather than 50 for each, it should have 
Columns: 131
Rows: 245
The runoff value should be a floating point, so rather than 1, set this to 1.0. 
I ran this and the result looked like the below figure. I did recompile the code on my own computer; I couldn't run your compilation since it used a different version of MPI. I don't think that should be a problem, but if this still doesn't work for you, we can investigate further.  
![Figure_1](https://user-images.githubusercontent.com/41647227/59734765-52f26b00-9218-11e9-8317-9a0b7a4d43dc.png)
